buildenv/jenkins/openjdk_tests

```diff
@@ -222,52 +228,88 @@ if (JDK_VERSIONS.size() > 1 || JDK_IMPLS.size() > 1 || PLATFORMS.size() >1 || PL
         println "SPEC: ${SPEC}"
         println "LABEL: ${LABEL}"
 
+        dynamicAgents = []
+        if (LABEL == PLATFORM_MAP[params.PLATFORM]["LABEL"] && PLATFORM_MAP[params.PLATFORM].containsKey('DynamicAgents')) {
+            dynamicAgents = PLATFORM_MAP[params.PLATFORM]["DynamicAgents"]
+        }
+
         stage('Queue') {
-            if (nodesByLabel(LABEL).size() < 1) {
+            String[] onlineNodes = nodesByLabel(LABEL)
+            if (onlineNodes.size() < 1) {
                 int ACTIVE_NODE_TIMEOUT = params.ACTIVE_NODE_TIMEOUT ? params.ACTIVE_NODE_TIMEOUT : 0
-
                 timeout(ACTIVE_NODE_TIMEOUT) {
-                    // node(LABEL) has to be used for spinning up a dynamic vm agent
+                    // If there is available node before timeout
                     node(LABEL) {
-                        echo "find the node ${env.NODE_NAME}"
+                        echo "find the node with label as ${env.NODE_NAME}"
                     }
                 }
-            }
+            } else {
+                // IF no nodes are idle we will check if there is supported virtual agent
+                // When Parallel the race condition could happen. Say the number of multiply jobs is larger than the available nodes the query's result may be delayed and wrong
+                // In this case jobs will be fooled to fall back to wait local busy nodes. 
+                if (params.CLOUD_PROVIDER != null && params.CLOUD_PROVIDER.length() != 0) {
```

Comment:
We only need to run the following logic if CLOUD_PROVIDER is also in dynamicAgents.